src/store/types.ts

```diff
@@ -0,0 +1,153 @@
+import { data } from '../data/data';
+import { PetsInterface } from '../data/types';
+import { events } from 'enums';
+import { FilterInterface } from '../components/filter/types';
+
+export abstract class ActiveFilterStoreInterface {
+    protected _activeFilters: FilterInterface;
+
+    constructor(activeFilters: FilterInterface) {
+        this._activeFilters = activeFilters;
+    }
+
+    add = (name: keyof FilterInterface, value: string) => {
+        const currentData = this._activeFilters[name];
+        if (currentData) {
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            this._activeFilters[name] = [...currentData, value];
+        } else {
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            this._activeFilters[name] = [value];
+        }
+    };
+
+    remove = (name: keyof FilterInterface, value: string) => {
+        const currentData = this._activeFilters[name];
+        if (currentData) {
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            const preparedData = currentData?.filter((el) => el !== value);
+            if (preparedData.length) {
+                this._activeFilters[name] = preparedData;
+            } else {
+                delete this._activeFilters[name];
+            }
+        }
+    };
+
+    removeAll = () => {
+        this._activeFilters = {};
+    };
+
+    get = () => {
+        return this._activeFilters;
+    };
+}
+
+enum storageKeys {
+    ORDER = 'order',
+}
+
+const StoreEvent = new CustomEvent(events.STORE_UPDATED);
+
+export abstract class CartStoreInterface {
+    protected _orderInfo: cartProductInterface[];
+
+    constructor() {
+        this._orderInfo = this.getStorage();
+    }
+
+    get = () => {
+        return this._orderInfo;
+    };
+
+    getQuantity = () => {
+        return this.get().reduce((acc, rec) => acc + rec.amount, 0);
+    };
+
+    getById = (productId: number) => {
+        const cuurentProduct = this.get().find(({ id }) => productId === id);
+        return cuurentProduct || null;
+    };
+
+    set = (orders: cartProductInterface[]) => {
+        this._orderInfo = orders;
+    };
+
+    increase = (orderId: number) => {
+        let currentProducts = this.get();
```

Comment:
Please try to avoid `let`

Reply:
I use currentProducts to avoid mutating the original array. The point is to return a new array and not mutate the old one.

Reply:
Use different consts for `currentProducts` and `newProducts`, try to use `? : ` for getting these new products